### PR TITLE
Булгаков - Лабораторная работа #1

### PR DESCRIPTION
**Pull request by danielbulgakov**

### CODE REVIEW

**Comment by danielbulgakov** [comment]
@TibidoX @Ilya837 ревью

**Comment by TibidoX** [comment]
very nice

File: modules/complex-number/test/test_BULGAKOV_DANIIL_complex_number.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 Bulgakov Daniil
+
+#include <gtest/gtest.h>
+
+#include "include/complex_number.h"
+
+TEST(Bulgakov_Daniil_ComplexNumberTest, Can_Multiple_Sum) {
+    // Arrange
+    ComplexNumber a(1.0, 1.0);
+    ComplexNumber expect(3.0, 3.0);
```
Comment (dkurt) [inline modules/complex-number/test/test_BULGAKOV_DANIIL_complex_number.cpp:10]: Values should be different

File: modules/complex-number/test/test_BULGAKOV_DANIIL_complex_number.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    ComplexNumber result = a + a + a;
+
+    // Assert
+    ASSERT_EQ(expect.getIm(), result.getIm());
+    ASSERT_EQ(expect.getRe(), result.getRe());
+}
+
+TEST(Bulgakov_Daniil_ComplexNumberTest, Test_Selfsubstruct_Zero) {
+    // Arrange
+    ComplexNumber a(1.0, 1.0);
```
Comment (dkurt) [inline modules/complex-number/test/test_BULGAKOV_DANIIL_complex_number.cpp:22]: Same as above

**Review by TibidoX — APPROVED** [review]
Very-very nice!

**Review by imterinaa — APPROVED** [review]
great =)

File: modules/complex-number/test/test_BULGAKOV_DANIIL_complex_number.cpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    // Assert
+    ASSERT_EQ(expect.getIm(), result.getIm());
+    ASSERT_EQ(expect.getRe(), result.getRe());
+}
+
+TEST(Bulgakov_Daniil_ComplexNumberTest, Test_Selfsubstruct_Zero) {
+    // Arrange
+    ComplexNumber a(12.5, 32.1);
+    ComplexNumber expect(0.0, 0.0);
```
Comment (dkurt) [inline modules/complex-number/test/test_BULGAKOV_DANIIL_complex_number.cpp:25]: Make non-zero

**Comment by danielbulgakov** [comment]
@dkurt how can you subtract a number from the exact same number without getting 0?

File: modules/complex-number/test/test_BULGAKOV_DANIIL_complex_number.cpp (hunk 3 above)
Comment (danielbulgakov) [reply modules/complex-number/test/test_BULGAKOV_DANIIL_complex_number.cpp:25]: How can you substract a number from the exact same number without getting 0?

Comment (dkurt) [reply modules/complex-number/test/test_BULGAKOV_DANIIL_complex_number.cpp:25]: Same question - same comment here: https://github.com/Teddy-Hackers/devtools-course-practice/pull/60#discussion_r1178685817

**Pull request merged**